File: src/main/java/br/com/svbgabriel/gerenciadortarefas/repositories/RepositoryTarefa.java
package br.com.svbgabriel.gerenciadortarefas.repositories;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import br.com.svbgabriel.gerenciadortarefas.models.Tarefa;

public interface RepositoryTarefa extends JpaRepository<Tarefa, Long> {

	@Query("SELECT t FROM Tarefa t WHERE t.usuario.email = :emailUsuario")
	List<Tarefa> carregarTarefasPorUsuario(@Param("emailUsuario") String email);

}

File: src/main/kotlin/br/com/svbgabriel/gerenciadortarefas/repositories/RepositoryTarefa.kt
package br.com.svbgabriel.gerenciadortarefas.repositories

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import br.com.svbgabriel.gerenciadortarefas.models.Tarefa

interface RepositoryTarefa : JpaRepository<Tarefa, Long> {
	@Query("SELECT t FROM Tarefa t WHERE t.usuario.email = :emailUsuario")
	fun carregarTarefasPorUsuario(@Param("emailUsuario") email: String): List<Tarefa>?
}